day3/day3.erl

```diff
@@ -0,0 +1,108 @@
+-module(day3).
+-export([find_closest_intersection/0, fewest_steps_to_intersection/0]).
+
+read_input(InputFile) ->
+    {'ok', Input} = file:read_file(InputFile),
+    binary:part(Input, {0, byte_size(Input) - 1}).
+
+process_input(Input) ->
+    [Binary1, Binary2] = binary:split(Input, <<"\n">>),
+    {binary:split(Binary1, <<",">>, ['global']), binary:split(Binary2, <<",">>, ['global'])}.
+
+binary_to_direction(<<"U">>) -> up;
+binary_to_direction(<<"D">>) -> down;
+binary_to_direction(<<"L">>) -> left;
+binary_to_direction(<<"R">>) -> right.
+
+binary_to_wiremap(L) ->
+    binary_to_wiremap(L, []).
+
+binary_to_wiremap([H | T], WireMap) ->
+    <<BinaryDirection:1/binary, BinaryDistance/binary>> = H,
+    binary_to_wiremap(T, [{binary_to_direction(BinaryDirection), binary_to_integer(BinaryDistance)} | WireMap]);
+binary_to_wiremap([], Directions) ->
+    lists:reverse(Directions).
+
+wiremap_to_wire_plot(L) ->
+    wiremap_to_wire_plot(L, [{0, 0}], {0, 0}).
+
+wiremap_to_wire_plot([{Direction, Distance} | T], Points, CurrentPoint) ->
+    {NewSegment, LastPoint} = add_line_segment(Direction, Distance, CurrentPoint, []),
+    wiremap_to_wire_plot(T, [NewSegment | Points], LastPoint);
+wiremap_to_wire_plot([], Points, _) ->
+    lists:flatten(lists:reverse(Points)).
+
+add_line_segment(_, 0, CurrentPoint, Points) ->
+    {lists:reverse(Points), CurrentPoint};
+add_line_segment('up', Distance, {X, Y}, Points) ->
+    add_line_segment('up', Distance - 1, {X, Y + 1}, [{X, Y + 1} | Points]);
+add_line_segment('down', Distance, {X, Y}, Points) ->
+    add_line_segment('down', Distance - 1, {X, Y - 1}, [{X, Y - 1} | Points]);
+add_line_segment('left', Distance, {X, Y}, Points) ->
+    add_line_segment('left', Distance - 1, {X - 1, Y}, [{X - 1, Y} | Points]);
+add_line_segment('right', Distance, {X, Y}, Points) ->
+    add_line_segment('right', Distance - 1, {X + 1, Y}, [{X + 1, Y} | Points]).
+
+find_intersections(L1, L2) ->
+    sets:to_list(sets:intersection(sets:from_list(L1), sets:from_list(L2))).
+
+manhatten_distance({X, Y}) ->
```

Comment:
This only works since you know the other point is the origin `{0,0}`. This is not the generalized version (just FYI - it works for this problem but isn't a generic equation for the distance).